terraform show -json of tfplan file crashes with panic: runtime error: invalid memory address or nil pointer dereference (the show without json works fine)

Hi,

I am getting an error when trying to view the json for my tfplan

### Terraform Version
```
Terraform v0.12.0
+ provider.aws v2.12.0
```

### Terraform Configuration Files
```hcl
module "my-module-1" {
  source                  = "git::...."
  ...
  sns_topic_arn           = module.my-module-2.this_arn
  ApproximateAgeOfOldestMessage = {}
  ApproximateNumberOfMessagesNotVisible = {}
  ApproximateNumberOfMessagesVisible = {}
  NumberOfMessagesDeleted = {}
  NumberOfMessagesReceived = {}
  NumberOfMessagesSent = {}
  SentMessageSize = {}
}
```
I have a module, that has a git source to another module which will create SQS queues, kms key, and cloudwatch metric alarms for the SQS queues. The terraform show outputs the following
```
  + module.my-module-2.aws_sns_topic.this

  + module.my-module-1.module.sqs.aws_sqs_queue.this[0]

  + module.my-module-1.module.sqs.aws_sqs_queue.this[1]

  + module.my-module-1.module.sqs.aws_sqs_queue.this[2]

  + module.my-module-1.module.sqs.aws_sqs_queue.this[3]

  + module.my-module-1.module.sqs.aws_sqs_queue_policy.this_sqs_policy[0]

  + module.my-module-1.module.sqs.aws_sqs_queue_policy.this_sqs_policy[1]

  + module.my-module-1.module.sqs.aws_sqs_queue_policy.this_sqs_policy[2]

  + module.my-module-1.module.sqs.aws_sqs_queue_policy.this_sqs_policy[3]

  + module.my-module-1.module.sqs_alarms.aws_cloudwatch_metric_alarm.sqs_age_of_oldest[0]

  + module.my-module-1.module.sqs_alarms.aws_cloudwatch_metric_alarm.sqs_age_of_oldest[1]

  + module.my-module-1.module.sqs_alarms.aws_cloudwatch_metric_alarm.sqs_age_of_oldest[2]

  + module.my-module-1.module.sqs_alarms.aws_cloudwatch_metric_alarm.sqs_age_of_oldest[3]

  + module.my-module-1.module.sqs_alarms.aws_cloudwatch_metric_alarm.sqs_delete_messages[0]

  + module.my-module-1.module.sqs_alarms.aws_cloudwatch_metric_alarm.sqs_delete_messages[1]

  + module.my-module-1.module.sqs_alarms.aws_cloudwatch_metric_alarm.sqs_delete_messages[2]

  + module.my-module-1.module.sqs_alarms.aws_cloudwatch_metric_alarm.sqs_delete_messages[3]

  + module.my-module-1.module.sqs_alarms.aws_cloudwatch_metric_alarm.sqs_in_flight[0]

  + module.my-module-1.module.sqs_alarms.aws_cloudwatch_metric_alarm.sqs_in_flight[1]

  + module.my-module-1.module.sqs_alarms.aws_cloudwatch_metric_alarm.sqs_in_flight[2]

  + module.my-module-1.module.sqs_alarms.aws_cloudwatch_metric_alarm.sqs_in_flight[3]

  + module.my-module-1.module.sqs_alarms.aws_cloudwatch_metric_alarm.sqs_message_size[0]

  + module.my-module-1.module.sqs_alarms.aws_cloudwatch_metric_alarm.sqs_message_size[1]

  + module.my-module-1.module.sqs_alarms.aws_cloudwatch_metric_alarm.sqs_message_size[2]

  + module.my-module-1.module.sqs_alarms.aws_cloudwatch_metric_alarm.sqs_message_size[3]

  + module.my-module-1.module.sqs_alarms.aws_cloudwatch_metric_alarm.sqs_receive_messages[0]

  + module.my-module-1.module.sqs_alarms.aws_cloudwatch_metric_alarm.sqs_receive_messages[1]

  + module.my-module-1.module.sqs_alarms.aws_cloudwatch_metric_alarm.sqs_receive_messages[2]

  + module.my-module-1.module.sqs_alarms.aws_cloudwatch_metric_alarm.sqs_receive_messages[3]

  + module.my-module-1.module.sqs_alarms.aws_cloudwatch_metric_alarm.sqs_sent_messages[0]

  + module.my-module-1.module.sqs_alarms.aws_cloudwatch_metric_alarm.sqs_sent_messages[1]

  + module.my-module-1.module.sqs_alarms.aws_cloudwatch_metric_alarm.sqs_sent_messages[2]

  + module.my-module-1.module.sqs_alarms.aws_cloudwatch_metric_alarm.sqs_sent_messages[3]

  + module.my-module-1.module.sqs_alarms.aws_cloudwatch_metric_alarm.sqs_visible_messages[0]

  + module.my-module-1.module.sqs_alarms.aws_cloudwatch_metric_alarm.sqs_visible_messages[1]

  + module.my-module-1.module.sqs_alarms.aws_cloudwatch_metric_alarm.sqs_visible_messages[2]

  + module.my-module-1.module.sqs_alarms.aws_cloudwatch_metric_alarm.sqs_visible_messages[3]

  + module.my-module-1.module.sqs_kms_key.aws_kms_alias.this_alias[0]

  + module.my-module-1.module.sqs_kms_key.aws_kms_key.this[0]
```

### Crash Output
https://gist.github.com/shahdk/d93e20ff0a65545791e85d9e3f715a25

### Expected Behavior
A valid json output of the terraform plan file

### Actual Behavior
Terraform crashed
```
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x30 pc=0x2290b98]

goroutine 1 [running]:
github.com/hashicorp/terraform/command/jsonconfig.marshalModuleCall(0x0, 0xc000468160, 0xc000794db0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, ...)
        /opt/gopath/src/github.com/hashicorp/terraform/command/jsonconfig/config.go:257 +0x1a8
github.com/hashicorp/terraform/command/jsonconfig.marshalModuleCalls(0xc00082bd40, 0xc000794db0, 0xc000519397)
        /opt/gopath/src/github.com/hashicorp/terraform/command/jsonconfig/config.go:234 +0x13b
github.com/hashicorp/terraform/command/jsonconfig.marshalModule(0xc00082bd40, 0xc000794db0, 0xc0001e58a0, 0x11, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, ...)
        /opt/gopath/src/github.com/hashicorp/terraform/command/jsonconfig/config.go:204 +0x589
github.com/hashicorp/terraform/command/jsonconfig.marshalModuleCall(0xc00082bd40, 0xc0001a9a20, 0xc000794db0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, ...)
        /opt/gopath/src/github.com/hashicorp/terraform/command/jsonconfig/config.go:264 +0x3a6
github.com/hashicorp/terraform/command/jsonconfig.marshalModuleCalls(0xc0003e5ee0, 0xc000794db0, 0xc0005e20a0)
        /opt/gopath/src/github.com/hashicorp/terraform/command/jsonconfig/config.go:234 +0x13b
github.com/hashicorp/terraform/command/jsonconfig.marshalModule(0xc0003e5ee0, 0xc000794db0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, ...)
        /opt/gopath/src/github.com/hashicorp/terraform/command/jsonconfig/config.go:204 +0x589
github.com/hashicorp/terraform/command/jsonconfig.Marshal(0xc0003e5ee0, 0xc000794db0, 0xc001124000, 0x20a6, 0x2500, 0x0, 0x0)
        /opt/gopath/src/github.com/hashicorp/terraform/command/jsonconfig/config.go:122 +0xbd
github.com/hashicorp/terraform/command/jsonplan.Marshal(0xc0003e5ee0, 0xc000325570, 0xc00001d470, 0xc000794db0, 0xc0005d9790, 0x1, 0x0, 0x0, 0x0, 0x0)
        /opt/gopath/src/github.com/hashicorp/terraform/command/jsonplan/plan.go:135 +0x598
github.com/hashicorp/terraform/command.(*ShowCommand).Run(0xc0004a4300, 0xc00003a0e0, 0x2, 0x2, 0xc0004122c0)
        /opt/gopath/src/github.com/hashicorp/terraform/command/show.go:167 +0xa58
github.com/mitchellh/cli.(*CLI).Run(0xc000492000, 0xc000492000, 0xc000265d90, 0x1)
        /opt/gopath/pkg/mod/github.com/mitchellh/cli@v1.0.0/cli.go:255 +0x1f1
main.wrappedMain(0x0)
        /opt/gopath/src/github.com/hashicorp/terraform/main.go:223 +0xaff
main.realMain(0x0)
        /opt/gopath/src/github.com/hashicorp/terraform/main.go:100 +0xb4
main.main()
        /opt/gopath/src/github.com/hashicorp/terraform/main.go:36 +0x3b



!!!!!!!!!!!!!!!!!!!!!!!!!!! TERRAFORM CRASH !!!!!!!!!!!!!!!!!!!!!!!!!!!!

Terraform crashed! This is always indicative of a bug within Terraform.
A crash log has been placed at "crash.log" relative to your current
working directory. It would be immensely helpful if you could please
report the crash with Terraform[1] so that we can fix this.

When reporting bugs, please include your terraform version. That
information is available on the first line of crash.log. You can also
get it by running 'terraform --version' on the command line.

[1]: https://github.com/hashicorp/terraform/issues

!!!!!!!!!!!!!!!!!!!!!!!!!!! TERRAFORM CRASH !!!!!!!!!!!!!!!!!!!!!!!!!!!!
```

--- a/command/jsonconfig/config.go
+++ b/command/jsonconfig/config.go
@@ -238,8 +238,7 @@ func marshalModuleCalls(c *configs.Config, schemas *terraform.Schemas) map[strin
 }
 
 func marshalModuleCall(c *configs.Config, mc *configs.ModuleCall, schemas *terraform.Schemas) moduleCall {
-	// It's possible, though unlikely, to have a module call with a nil config
-	// https://github.com/hashicorp/terraform/issues/21543
+	// It is possible to have a module call with a nil config.
 	if c == nil {
 		return moduleCall{}
 	}

--- a/command/jsonconfig/config.go
+++ b/command/jsonconfig/config.go
@@ -238,6 +238,12 @@ func marshalModuleCalls(c *configs.Config, schemas *terraform.Schemas) map[strin
 }
 
 func marshalModuleCall(c *configs.Config, mc *configs.ModuleCall, schemas *terraform.Schemas) moduleCall {
+	// It's possible, though unlikely, to have a module call with a nil config
+	// https://github.com/hashicorp/terraform/issues/21543
+	if c == nil {
+		return moduleCall{}
+	}
+
 	ret := moduleCall{
 		Source:            mc.SourceAddr,
 		VersionConstraint: mc.Version.Required.String(),

--- a/command/test-fixtures/show-json/nested-modules/main.tf
+++ b/command/test-fixtures/show-json/nested-modules/main.tf
@@ -1,0 +1,3 @@
+module "my_module" {
+  source = "./modules"
+}

--- a/command/test-fixtures/show-json/nested-modules/modules/main.tf
+++ b/command/test-fixtures/show-json/nested-modules/modules/main.tf
@@ -1,0 +1,3 @@
+module "more" {
+  source = "./more-modules"
+}

--- a/command/test-fixtures/show-json/nested-modules/modules/more-modules/main.tf
+++ b/command/test-fixtures/show-json/nested-modules/modules/more-modules/main.tf
@@ -1,0 +1,4 @@
+variable "misspelled" {
+  default     = "ehllo"
+  descriptoni = "I am a misspelled attribute"
+}

--- a/command/test-fixtures/show-json/nested-modules/output.json
+++ b/command/test-fixtures/show-json/nested-modules/output.json
@@ -1,0 +1,23 @@
+{
+    "format_version": "0.1",
+    "terraform_version": "0.12.1-dev",
+    "planned_values": {
+        "root_module": {}
+    },
+    "configuration": {
+        "root_module": {
+            "module_calls": {
+                "my_module": {
+                    "source": "./modules",
+                    "module": {
+                        "module_calls": {
+                            "more": {
+                                "module": {}
+                            }
+                        }
+                    }
+                }
+            }
+        }
+    }
+}

